src/writer/untabber.rs

```diff
@@ -0,0 +1,49 @@
+use std::io::{Result, Write};
+use super::TAB;
+
+pub struct Untabber {
+    orig: Box<Write>,
+}
+
+impl Untabber {
+    pub fn new(orig: Box<Write>) -> Untabber {
+        Untabber{ orig: orig }
+    }
+}
+
+impl Write for Untabber {
+    fn write(&mut self, buf: &[u8]) -> Result<usize> {
+        let v = buf.into();
+        match String::from_utf8(v) {
+            Ok(s) => {
+                let bytes = untab(&s).into_bytes();
+                let delta_len = bytes.len() - buf.len();
+                match self.orig.write(&bytes) {
+                    Ok(n) if n < delta_len => Ok(n),
+                    //hack for return normal len
+                    Ok(n) => Ok(n - delta_len),
```

Comment:
Code points longer than one byte have the highest bit set so we can be certain any `<128` characters are what they seem to be.

I realize now i should've used `write_all` there (and avoid writes if `chunk.is_empty()`) but since we're writing to files I believe we can ignore any exotic corner cases here and can just return `Ok(buf.len())` on success.